### PR TITLE
TLSv1.3: Don't check the record version

### PR DESCRIPTION
**Pull request by mattcaswell**
##### Checklist

- [x] tests are added or updated

##### Description of change

The record layer version field must be ignored in TLSv1.3, so we remove the check when using that version.

### CODE REVIEW

File: test/recipes/70-test_sslrecords.t [hunk 1]
```diff
@@ -388,3 +406,15 @@ sub add_unknown_record_type
...
+sub change_version
+{
+    my $proxy = shift;
+
+    # We'll change a version after the initial version neg has taken place
+    if ($proxy->flight != 2) {
+        return;
+    }
+
+    (${$proxy->record_list}[-1])->version(TLSProxy::Record::VERS_TLS_1_1);
```
Comment (richsalz) [inline test/recipes/70-test_sslrecords.t]: you have to be kidding me.  talk about write-once.

**Comment by mattcaswell** [comment]
Merged. Thanks.

**Comment by kaduk** [comment]
Where does the TLS 1.3 spec say that we must ignore the record version?
Looking at the -18, section 5.1 seems to say that it must be 0x0301 as part of the structure definition, so (implicitly) we should throw a decode error if we receive a different version.

**Comment by mattcaswell** [comment]
````
   legacy_record_version  This value MUST be set to 0x0301 for all
      records.  This field is deprecated and MUST be ignored for all
      purposes.
````

**Comment by kroeckx** [comment]
I was also wondering if we should check that it was 0x0301, but then I looked it up and it clearly says to ignore it.

**Comment by kroeckx** [comment]
And I sort of have a problem with this, in that Postel was wrong. If the other side has a MUST send that, I think we SHOULD check it.

**Comment by kaduk** [comment]
It looks like Mint checks it; I haven't looked at Boring/NSS/etc. (yet?).

**Comment by kaduk** [comment]
Assuming that I found the right spot (tls_open_record()), Boring seems to just check the first octet.
NSS (tls13_UnprotectRecord()) seems to check both octets.

**Comment by mattcaswell** [comment]
I'd suggest you raise it on the tls list.

**Comment by kaduk** [comment]
> I'd suggest you raise it on the tls list.

I was already composing as you sent that; you are CC'd.

**Pull request closed without merging**